pkg/controller/infrastructure/actuator_reconcile.go

```diff
@@ -41,11 +41,7 @@ import (
 )
 
 func (a *actuator) Reconcile(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure, _ *extensionscontroller.Cluster) error {
-	infrastructureStatus, state, err := Reconcile(ctx, a.logger, a.RESTConfig(), a.Client(), a.Decoder(), a.ChartRenderer(), infrastructure)
```

Comment:
Sorry @vlvasilev, I forgot that the reason for this construct here was that we are using it in the integration tests for the infrastructure. Hence, my advise to update the provider status inside the `Reconcile()` function was wrong. Can you please revert it and adapt the `Restore` function accordingly (like you had it done before)?

Reply:
ok